packages/search/src/search.tsx

```diff
@@ -90,349 +96,349 @@ type State = {
+	const searchValue = keyword;
+	const placeholder = placeholderProp || __( 'Search…', __i18n_text_domain__ );
+	const isOpenUnpinnedOrQueried = isOpen || ! pinned || searchValue;
 
-		const autocorrectProps = this.props.disableAutocorrect && {
-			autoComplete: 'off',
-			autoCorrect: 'off',
-			spellCheck: 'false' as const,
-		};
+	const autocorrectProps = disableAutocorrect && {
+		autoComplete: 'off',
+		autoCorrect: 'off',
+		spellCheck: 'false' as const,
+	};
 
-		const searchClass = classNames( 'search-component', this.props.className, this.props.dir, {
-			'is-expanded-to-container': this.props.fitsContainer,
-			'is-open': isOpenUnpinnedOrQueried,
-			'is-searching': this.props.searching,
-			'is-compact': this.props.compact,
-			'has-focus': this.state.hasFocus,
-			'has-open-icon': ! this.props.hideOpenIcon,
-		} );
+	const searchClass = classNames( 'search-component', className, dir, {
+		'is-expanded-to-container': fitsContainer,
+		'is-open': isOpenUnpinnedOrQueried,
+		'is-searching': searching,
+		'is-compact': compact,
+		'has-focus': hasFocus,
+		'has-open-icon': ! hideOpenIcon,
+	} );
 
-		const fadeClass = classNames( 'search-component__input-fade', this.props.dir );
-		const inputClass = classNames( 'search-component__input', this.props.dir );
+	const fadeClass = classNames( 'search-component__input-fade', dir );
+	const inputClass = classNames( 'search-component__input', dir );
 
-		const shouldRenderRightOpenIcon = this.props.openIconSide === 'right' && ! this.state.keyword;
+	const shouldRenderRightOpenIcon = openIconSide === 'right' && ! keyword;
 
-		return (
-			<div dir={ this.props.dir } className={ searchClass } role="search">
-				<Spinner />
-				{ this.props.openIconSide === 'left' && this.renderOpenIcon() }
-				<form className={ fadeClass } action="." onSubmit={ this.handleSubmit }>
-					<input
-						type="search"
-						id={ 'search-component-' + this.instanceId }
-						autoFocus={ this.props.autoFocus } // eslint-disable-line jsx-a11y/no-autofocus
-						aria-describedby={ this.props.describedBy }
-						aria-label={ inputLabel ? inputLabel : this.props.__( 'Search', __i18n_text_domain__ ) }
-						aria-hidden={ ! isOpenUnpinnedOrQueried }
-						className={ inputClass }
-						placeholder={ placeholder }
-						role="searchbox"
-						value={ searchValue }
-						ref={ this.searchInput }
-						onChange={ this.onChange }
-						onKeyUp={ this.keyUp }
-						onKeyDown={ this.keyDown }
-						onMouseUp={ this.props.onClick }
-						onFocus={ this.onFocus }
-						onBlur={ this.onBlur }
-						disabled={ this.props.disabled }
-						autoCapitalize="none"
-						dir={ this.props.dir }
-						maxLength={ this.props.maxLength }
-						minLength={ this.props.minLength }
-						{ ...autocorrectProps }
-					/>
-					{ this.renderStylingDiv() }
-				</form>
-				{ shouldRenderRightOpenIcon ? this.renderOpenIcon() : this.closeButton() }
-			</div>
-		);
-	}
+	const renderStylingDiv = React.useCallback( () => {
```

Comment:
None of the `render*` function needs to be wrapped/memoized with `useCallback`. They are called in a way where it doesn't matter it's a different instance on every call. We can simplify the code by writing just
```js
function renderStylingDiv() {
  ...
}
```

Reply:
I don't really understand why this is the case, can you elaborate more?

Reply:
You're not passing the `render` functions to any component as props. They are merely local helper functions that are called only locally during render:
```js
function MyComponent() {
  function renderPart() {
    return <div>hello</div>;
  }

  return <div>{ renderPart() }</div>;
}
```
Memoizing this with `useCallback` doesn't provide any benefit.

Reply:
Oh, of course, because the function will be re-declared on each render anyway 🤦🏼‍♀️ I get it now, thanks!